contracts/utils/Votes.sol

```diff
@@ -0,0 +1,210 @@
+        emit DelegateChanged(account, oldDelegation, newDelegation);
+
+        _moveVotingPower(oldDelegation, newDelegation, balance);
+    }
+
+    /**
+     * @dev Moves voting power.
+     */
+    function _moveVotingPower(
+        address src,
+        address dst,
+        uint256 amount
+    ) internal {
+        if (src != dst && amount > 0) {
+            if (dst == address(0) && src != address(0)) {
+                _totalCheckpoints.push(_subtract, amount);
+            } else if (src == address(0) && dst != address(0)) {
+                _totalCheckpoints.push(_add, amount);
+            }
+
+            if (src != address(0)) {
+                (uint256 oldValue, uint256 newValue) = _userCheckpoints[src].push(_subtract, amount);
+                emit DelegateVotesChanged(src, oldValue, newValue);
+            }
+            if (dst != address(0)) {
+                (uint256 oldValue, uint256 newValue) = _userCheckpoints[dst].push(_add, amount);
+                emit DelegateVotesChanged(dst, oldValue, newValue);
+            }
+        }
+    }
+
+    /**
+     * @dev Delegates votes from signer to `delegatee`
+     */
+    function delegateBySig(
+        address delegatee,
+        uint256 nonce,
+        uint256 expiry,
+        uint8 v,
+        bytes32 r,
+        bytes32 s
+    ) public virtual {
+        require(block.timestamp <= expiry, "ERC721Votes: signature expired");
+        address signer = ECDSA.recover(
+            _hashTypedDataV4(keccak256(abi.encode(_DELEGATION_TYPEHASH, delegatee, nonce, expiry))),
+            v,
+            r,
+            s
+        );
+        require(nonce == _useNonce(signer), "ERC721Votes: invalid nonce");
+        _delegate(signer, delegatee, _getDelegatorVotes(signer));
+    }
+
+    /**
+     * @dev Adds two numbers.
+     */
+    function _add(uint256 a, uint256 b) private pure returns (uint256) {
+        return a + b;
+    }
+
+    /**
+     * @dev Subtracts two numbers.
+     */
+    function _subtract(uint256 a, uint256 b) private pure returns (uint256) {
+        return a - b;
+    }
+
+    /**
+     * @dev "Consume a nonce": return the current value and increment.
+     *
+     * _Available since v4.1._
+     */
+    function _useNonce(address owner) internal virtual returns (uint256 current) {
+        Counters.Counter storage nonce = _nonces[owner];
+        current = nonce.current();
+        nonce.increment();
+    }
+
+    /**
+     * @dev Returns an address nonce.
+     */
+    function nonces(address owner) public view virtual returns (uint256) {
+        return _nonces[owner].current();
+    }
+
+    /**
+     * @dev Returns DOMAIN_SEPARATOR.
+     */
+    // solhint-disable-next-line func-name-mixedcase
+    function DOMAIN_SEPARATOR() external view returns (bytes32) {
+        return _domainSeparatorV4();
+    }
+
+    /**
+     * @dev Returns the balance of the delegator account
+     */
+    function _getDelegatorVotes(address) internal virtual returns (uint256);
```

Comment:
I think we should use the term "voting power" instead of "votes" here. In that way it naturally looks like a counterpart to `moveVotingPower`.

Reply:
fixed on `f054441f`